src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt

```diff
@@ -0,0 +1,62 @@
+package app.revanced.patches.youtube.interaction.copyvideourl.bytecode.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.youtube.interaction.copyvideourl.annotation.CopyVideoUrlCompatibility
+import app.revanced.patches.youtube.interaction.copyvideourl.resource.patch.CopyVideoUrlResourcePatch
+import app.revanced.patches.youtube.misc.playercontrols.bytecode.patch.PlayerControlsBytecodePatch
+import app.revanced.patches.youtube.misc.video.information.patch.VideoInformationPatch
+import app.revanced.patches.youtube.misc.video.videoid.patch.VideoIdPatch
+
+@Patch
+@Name("copy-video-url")
+@Description("Adds buttons in player to copy video links.")
+@DependsOn([
+    CopyVideoUrlResourcePatch::class,
+    PlayerControlsBytecodePatch::class,
+    VideoIdPatch::class,
+    VideoInformationPatch::class
+])
+@CopyVideoUrlCompatibility
+@Version("0.0.1")
+class CopyVideoUrlBytecodePatch : BytecodePatch() {
+    private companion object {
+        const val INTEGRATIONS_PACKAGE = "Lapp/revanced/integrations"
+        const val INTEGRATIONS_PLAYER_PACKAGE = "$INTEGRATIONS_PACKAGE/videoplayer"
+        const val INTEGRATIONS_PATCH_UTILS_DESCRIPTOR = "$INTEGRATIONS_PACKAGE/patches/CopyVideoUrlPatch;"
+        val BUTTONS_DESCRIPTORS = listOf(
+            "$INTEGRATIONS_PLAYER_PACKAGE/CopyVideoUrlButton;",
+            "$INTEGRATIONS_PLAYER_PACKAGE/CopyVideoUrlTimestampButton;"
+        )
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        // Hook video time
+        with(VideoInformationPatch) {
+            videoTimeHook(
+                INTEGRATIONS_PATCH_UTILS_DESCRIPTOR,
+                "setVideoTime"
+            )
+        }
+
+        // Inject call for video id
+        VideoIdPatch.injectCall("$INTEGRATIONS_PATCH_UTILS_DESCRIPTOR->setVideoId(Ljava/lang/String;)V")
```

Comment:
This is not necessary. There is another patch that can be used as a dependency for this one. The integrations of that patch contain a getter for the current video id. The only time `VideoIdPatch` is required is if you need to update something based on a new video or its id. For example, the `return-youtube-dislike` requires it, in order to update the dislike count. This PR adds buttons that only require to use of the mentioned getter for the video id.

Reply:
Integrations of some patches (RYD, remember video quality and downloads) has getter method for `videoId`, but I couldn't find any other patch this patch can depend on.

Reply:
What I meant is that the `VideoInformation` patch should include a getter for the videoId. The `downloads` and the patch from this PR would then make use of it. For that the field `videoId` from the `downloads` patch should be moved to
 `app.revanced.integrations.patches.VideoInformation`. Make sure the respective changes are also reflected in the `VideoInformation` patch (such as adding `VideoIdPatch.injectCall` to it and making the `downloads` and your patch dependant on the `VideoInformation` patch).